python/nx-cugraph/nx_cugraph/algorithms/centrality/betweenness.py

```diff
@@ -18,7 +18,12 @@
 __all__ = ["betweenness_centrality", "edge_betweenness_centrality"]
 
 
-@networkx_algorithm
+@networkx_algorithm(
+    is_incomplete=True,  # weight not supported
+    is_different=True,  # RNG with seed is different
+    plc="betweenness_centrality",
+    version_added="23.10",
```

Comment:
My initial reaction was to think it's kind of odd to be sprinkling metadata throughout the source code like this (I was initially envisioning a single yaml file or some other kind of text file that served as both the input to the tree/table dumping tools, and a doc itself that humans could easily refer to in a pinch)...however, the locality of the metadata to the implementation seems like it could be handy for nx-cugraph devs.

One thing I don't like is how the metadata mixes with the more critical args to the decorator (`extra_params`, `nodes_or_number`), making me think the implementation is using these new params at runtime somehow (or is/will it?).  IOW, it's not obvious that it's just metadata.  Perhaps this could instead be done with a single new param named `metadata` which is a dictionary?  That would also allow for easy expansion later and support for algo-specific metadata.
```suggestion
    metadata=dict(
        is_incomplete=True,  # weight not supported
        is_different=True,  # RNG with seed is different
        plc="betweenness_centrality",
        version_added="23.10",
    ),
```
As for the metadata itself, here's my thoughts:
* `is_incomplete`: This seems useful.  Although, it's really helpful to have the comment you added too ("# weight not supported").  What if it was a string or list of strings instead, where empty string or None indicated completeness?  Maybe then it could be renamed something like "missing_features", "not_supported", "not_supported_features", "lacuna", ...?
* `is_different`: This also seems useful, but again, even more so if it was a string/list of strings. Seems like it could also be named "quirks".
* `plc`: I'm starting to think this isn't worthy of the space and maintenance burden. It's only useful to a couple of cugraph devs, and  I'm guessing even that's debatable. PLC algo was initially added to the README table simply because it was an easy way to get a list (just dump all the PLC algos and make a table), but I think readers mainly care about what NX algo is supported, not what PLC algos get called.
* `version_added`: This one seems useful as-is.

Reply:
Thanks for the suggestions! I added clarifying code comments that emphasize that these are metadata only. I like them as keyword arguments--seems cleaner, easier to check, etc. Dict could be nice if we expected to have lots of different metadata fields.

Also, I like `is_*`, which are clearly boolean, and, if True, indicate you should look at the extra docstring to see details.